crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -109,6 +114,30 @@ impl TransactionBlock {
         ))
     }
 
+    /// Events emitted by this transaction block.
+    async fn events(
+        &self,
+        ctx: &Context<'_>,
+        first: Option<u64>,
+        after: Option<String>,
+        last: Option<u64>,
+        before: Option<String>,
+        filter: Option<EventFilter>,
+    ) -> Result<Option<Connection<String, Event>>> {
+        let mut event_filter = match filter {
+            Some(filter) => filter,
+            None => EventFilter::default(),
+        };
+
+        // Overwrite with the current transaction's digest.
+        event_filter.transaction_digest = Some(Base58::encode(&self.stored.transaction_digest));
```

Comment:
I believe you should return an empty connection if this field is set to some different value, to be completely consistent.

Reply:
ah yeah that makes sense